### PR TITLE
Added Single Video Details Page

### PR DESCRIPTION
**Pull request by sanjitsarkar**
**In this PR,**
I have added video info showing functionality.
Preview Link [Video Info Page](https://deploy-preview-4--zentube.netlify.app/)
**Things to check**
- [ ] SinglePageVideo

Things not to check
- [ ] Add to Playlist and Watch later and video liking features.

### CODE REVIEW

**Review by anshulraheja — APPROVED** [review]
Great UI. Code looks good to merge

File: src/components/Toast/index.jsx [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+  const [hide, setHide] = useState(true);
+  useEffect(() => {
+    if (show) {
+      setHide(false);
+      setTimeout(() => {
+        setHide(true);
+        toggleToast();
+      }, duration);
+    }
+  }, [show]);
```
Comment (iRohitGaur) [inline src/components/Toast/index.jsx:15]: suggestion: cleanup timeout

**Review by iRohitGaur — APPROVED** [review]
suggestion: make smaller PR

Nice work! Good to merge 👍🏻

File: src/components/Toast/index.jsx (hunk 1 above)
Comment (sanjitsarkar) [reply src/components/Toast/index.jsx:15]: Yes I have to cleanup timeout

**Review by zoyatanweer — APPROVED** [review]
clean code and easily understandable, good to merge.

File: src/pages/SingleVideoPage/index.jsx [hunk 2]
```diff
@@ -1,10 +1,44 @@
...
-
+import Loader from "../../components/Loader";
+import { useVideos } from "../../context/VideosContext";
+import RelatedVideos from "./RelatedVideos";
+import VideoInfo from "./VideoInfo";
+import "./SingleVideoPage.css";
 const SingleVideoPage = () => {
+  const location = useLocation();
+  const { fetchVideoInfo, video } = useVideos();
+  useEffect(async () => {
```
Comment (manish-GH) [inline src/pages/SingleVideoPage/index.jsx]: As far as I know async should not be used in useEffect like this. Check [this](https://devtrium.com/posts/async-functions-useeffect)

File: src/pages/SingleVideoPage/RelatedVideos.jsx [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+
+const RelatedVideos = ({ category, videoId }) => {
+  const { filterVideos, videos } = useVideos();
+  useEffect(async () => {
+    await filterVideos({ category });
+  }, [category]);
+  return (
+    <div className="mt-3">
+      <h3>Related Videos</h3>
+      <button className={`btn font-bold  btn-primary tag mt-2`}>
```
Comment (manish-GH) [inline src/pages/SingleVideoPage/RelatedVideos.jsx]: What is this button for? If there is no click action associated with it, I'd recommend you use some other tag

**Review by manish-GH — COMMENTED** [review]
The code is clean and easy to read. Nice work. 

Just have a look at the button and the useEffect. Then you are good to merge

File: src/pages/SingleVideoPage/RelatedVideos.jsx (hunk 3 above)
Comment (sanjitsarkar) [reply src/pages/SingleVideoPage/RelatedVideos.jsx]: Thanks for pointing out

File: src/pages/SingleVideoPage/index.jsx (hunk 2 above)
Comment (sanjitsarkar) [reply src/pages/SingleVideoPage/index.jsx]: Will fix

**Pull request merged**